src/NuGet.Core/NuGet.Protocol/docs/README.md

```diff
@@ -0,0 +1 @@
+NuGet's implementation for interacting with feeds. Contains functionality for all feed types.
```

Comment:
````suggestion
# NuGet.Protocol

Install the `NuGet.Protocol` package to interact with HTTP and folder-based NuGet package feeds. Learn more using [`NuGet.Protocol`'s documentation](https://docs.microsoft.com/nuget/reference/nuget-client-sdk#nugetprotocol) or [NuGet Server API's documentation](https://docs.microsoft.com/nuget/api/overview).

## Examples

### List package versions

Find all versions of Newtonsoft.Json:

```cs
ILogger logger = NullLogger.Instance;
CancellationToken cancellationToken = CancellationToken.None;

SourceCacheContext cache = new SourceCacheContext();
SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
FindPackageByIdResource resource = await repository.GetResourceAsync<FindPackageByIdResource>();

IEnumerable<NuGetVersion> versions = await resource.GetAllVersionsAsync(
    "Newtonsoft.Json",
    cache,
    logger,
    cancellationToken);

foreach (NuGetVersion version in versions)
{
    Console.WriteLine($"Found version {version}");
}
```

### Download a package

Download Newtonsoft.Json v12.0.1:

```cs
ILogger logger = NullLogger.Instance;
CancellationToken cancellationToken = CancellationToken.None;

SourceCacheContext cache = new SourceCacheContext();
SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
FindPackageByIdResource resource = await repository.GetResourceAsync<FindPackageByIdResource>();

string packageId = "Newtonsoft.Json";
NuGetVersion packageVersion = new NuGetVersion("12.0.1");
using MemoryStream packageStream = new MemoryStream();

await resource.CopyNupkgToStreamAsync(
    packageId,
    packageVersion,
    packageStream,
    cache,
    logger,
    cancellationToken);

Console.WriteLine($"Downloaded package {packageId} {packageVersion}");

using PackageArchiveReader packageReader = new PackageArchiveReader(packageStream);
NuspecReader nuspecReader = await packageReader.GetNuspecReaderAsync(cancellationToken);

Console.WriteLine($"Tags: {nuspecReader.GetTags()}");
Console.WriteLine($"Description: {nuspecReader.GetDescription()}");
```
````